asynctest/templates.nim

```diff
@@ -1,3 +1,18 @@
+template launderExceptions(body: typed) =
+  ## Chronos V4 requires that all procs which raise Exception annotate it
+  ## with {.async: (raises: [Exception]).}, but this construct does not
+  ## exist in asyncdispatch. We therefore launder all real instances of
+  ## Exception into CatchableError and make Chronos happy while not losing
+  ## context information for the remainder of the exception types.
+  try:
+    body
+  except Defect as ex:
+    raise ex
+  except CatchableError as ex:
+    raise ex
+  except Exception as ex:
```

Comment:
This should raise a `Defect`, not a `CatchableError` - any exception we can't track should be fatal.

Reply:
Fixed in 1ef3d363896c63aef3799e7b124ae88e76b172cb